src/error.ts

```diff
@@ -0,0 +1,47 @@
+interface FirebaseErrorOptions {
+  children?: unknown[];
+  context?: unknown;
+  exit?: number;
+  original?: Error;
+  status?: number;
+}
+
+const DEFAULT_CHILDREN: NonNullable<FirebaseErrorOptions["children"]> = [];
+const DEFAULT_EXIT: NonNullable<FirebaseErrorOptions["exit"]> = 1;
+const DEFAULT_STATUS: NonNullable<FirebaseErrorOptions["status"]> = 500;
+
+export class FirebaseError extends Error implements FirebaseErrorOptions {
```

Comment:
I think `implements FirebaseErrorOptions` is not appropriate. `FirebaseErrorOptions` is not meant to be an interface for a class, as far as I can tell.

Reply:
`FirebaseErrorOptions` are all options that can be specified on `FirebaseError` class, so enforcing existence of all in the class seems reasonable. I don't have a strong opinion on this, so I can change it, but it adds more safety and seems logical, at least to me.

Reply:
This is using an "options" interface to enforce the values on a "class", which I think is simply incorrect - it doesn't make sense to tie these two together. Let's remove the `implements`.

Reply:
54b491ad605b9bc1b6eff4e2ec54d09a29ce405e